File: Bit Manipulation/190. Reverse Bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) 
    {
        string bits = bitset<32>(n).to_string();
        reverse(bits.begin(), bits.end());

        int ans = stoll(bits, NULL, 2);
        return ans;
    }
};

File: Bit Manipulation/190. Reverse Bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        return int(bin(n)[2:].zfill(32)[::-1], 2)
    

class Solution:
    def reverseBits(self, n: int) -> int:
        bits = bin(n)[2:]
        bits = bits.zfill(32)
        bits = bits[::-1]

        return int(bits, 2)